developer/tutorial/extending-product-schema-location-map.md

```diff
@@ -109,62 +109,65 @@ The next step is to crate a new React component which is going to render the Goo
-     }
-   }
- }
-
- render() {
-   return (
-     <div>
-       <h3>{i18next.t("availableLocations", "Available at the following stores")}</h3>
-       <div className="map" ref="map" />
-     </div>
-   );
- }
+    product: PropTypes.object,
+    trackingId: PropTypes.string.isRequired
+  };
+
+  componentDidMount() {
+    if (this.props.trackingId) {
+      if ($("#mapsHead").length === 0) {
+        const url = `https://maps.googleapis.com/maps/api/js?key=${this.props.trackingId}`;
+        loadScript(url, { attrs: { id: "mapsHead" } }, () => {
+          this.renderMap();
+        });
+      } else {
+        this.renderMap();
+      }
+    }
+  }
+
+  renderMap() {
+    // eslint-disable-next-line no-undef, no-new
+    const googleMap = new google.maps.Map(this.refs.map, {
+      center: {
+        lat: this.props.product.lat,
+        lng: this.props.product.lng
+      },
+      zoom: 13
+    });
+
+    // eslint-disable-next-line no-undef, no-new
+    new google.maps.Marker({
+      position: {
+        lat: this.props.product.lat,
+        lng: this.props.product.lng
+      },
+      map: googleMap,
+      title: i18next.t("buyHere", "Buy here!")
+    });
+  }
+
+  render() {
+    return (
+      <div>
+        <h3>{i18next.t("availableLocations", "Available at the following stores")}</h3>
+        <div className="map" ref="map" />
+      </div>
+    );
+  }
 }
 
 export default AvailabilityMap;
 ```
 
-Great. This React component will inject the JavaScript we need and render the marker according our new product coordinates. One nice thing to notice is the fact, that ReactionCommerce's internal machinery will call our React component with appropriate context, namely the product itself. Therefor we get the React property `this.props.product` for free, which essentially is our document from database that features `lng` and `lat` information. What isn't provided out-of-the-box is the `trackingId` property needed for Google maps inclusion. This is your personal Google API key that is available from [developer.google.com](https://developers.google.com/maps/documentation/javascript/get-api-key). We're going to store that in our settings file in /settings/dev.settings.json:
+Great. This React component will inject the JavaScript we need and render the marker according our new product coordinates. One nice thing to notice is the fact, that ReactionCommerce's internal machinery will call our React component with appropriate context, namely the product itself. Therefore we get the React property `this.props.product` for free, which essentially is our document from database that features `lng` and `lat` information. What isn't provided out-of-the-box is the `trackingId` property needed for Google maps inclusion. This is your personal Google API key that is available from [developer.google.com](https://developers.google.com/maps/documentation/javascript/get-api-key). We're going to store that in our settings file in /settings/dev.settings.json:
```

Comment:
@machikoyasuda Added an 'e' to 'therefor'. Just in case you can't spot it immediately.